velox/functions/sparksql/String.h

```diff
@@ -583,4 +583,122 @@ struct OverlayVarbinaryFunction : public OverlayFunctionBase {
   }
 };
 
+/// translate(string, match, replace) -> varchar
+///
+///   Returns a new translated string. It translates the character in ``string``
+///   by a character in ``replace``. The character in ``replace`` is
+///   corresponding to the character in ``match``. The translation will
+///   happen when any character in ``string`` matching with the character in the
+///   ``match``.
+template <typename T>
+struct TranslateFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  // ASCII input always produces ASCII result.
+  static constexpr bool is_default_ascii_behavior = true;
+
+  FOLLY_ALWAYS_INLINE static std::function<int(const char*)> utf8CharCounter() {
+    return [](const char* input) { return utf8proc_char_length(input); };
+  }
+  FOLLY_ALWAYS_INLINE static std::function<int(const char*)>
+  asciiCharCounter() {
+    return [](const char*) { return 1; };
+  }
+
+  using MapType = folly::F14FastMap<StringView, StringView>;
```

Comment:
StringView is just a pointer. If the memory referenced by it is deallocated, this map will get corrupted.

For all-ASCII input, this can be just a map from char to char (or a plan array indexed by ASCII code of 'char')). For non-ASCII input you may need to store std::strings.

Reply:
> StringView is just a pointer. If the memory referenced by it is deallocated, this map will get corrupted.

Hi @mbasmanova, I think the map's key/value can only be accessed before the referred memory (for `match`/`replace`) is deallocated. Is there any possible case I am missing? Thanks!

Reply:
It is not guaranteed that StringViews passed to initialize() stay alive for as along as the function is being used. These can go away.